hel_django_auditlog_extra/README.md

```diff
@@ -0,0 +1,330 @@
+    audit log context before the request is processed. This ensures that
+    audit logs accurately reflect the user responsible for each action.
+
+    This fix is based on the suggestion provided in:
+    https://github.com/jazzband/django-auditlog/issues/115#issuecomment-1682234986
+
+    Additionally, this middleware sets the request path in the audit log
+    context, providing more context for each logged action.
+
+To use, add `"hel_django_auditlog_extra.middleware.AuditlogMiddleware"` to the list of the middlewares in `settings.py`, (instead of the one that `django-auditlog` offers):
+
+```python
+MIDDLEWARE = [
+    "django.middleware.security.SecurityMiddleware",
+    "django.contrib.auth.middleware.AuthenticationMiddleware",
+    # ...other middlewares that manipulates request context...
+    "hel_django_auditlog_extra.middleware.AuditlogMiddleware",
+]
+```
+
+### Graphene Decorators
+
+Code reference: [graphene_decorators.py](./graphene_decorators.py).
+
+#### `auditlog_access`
+
+    Decorator to init audit logging to a Graphene DjangoObjectType's get_node method.
+
+    Uses the `accessed` signal to log the access of the node.
+
+To use this decorator for a GraphQL Node, add it to the Node-class that is implementing a `DjangoObjectType`. The decorator will then wrap the `get_node` -function, that is inherited from the `DjangoObjectType`.
+
+```python
+@auditlog_access
+class ChildNode(DjangoObjectType):
+
+    # fields...
+
+    class Meta:
+        model = Child
+        # meta...
+
+    # methods...
+    @classmethod
+    @login_required
+    def get_node(cls, info, id):
+        try:
+            return cls._meta.model.objects.user_can_view(info.context.user).get(id=id)
+        except cls._meta.model.DoesNotExist:
+            return None
+```
+
+### Mixins
+
+Code reference: [mixins.py](./mixins.py).
+
+#### `AuditlogAdminViewAccessLogMixin`
+
+    A mixin for Django Admin views to log access events using `django-auditlog`.
+
+    This mixin automatically logs accesses to the `change_view`, `history_view`,
+    and `delete_view` in the Django Admin. It also provides an option to log
+    accesses to the `changelist_view` (list view).
+
+    By default, only access to individual object views (change, history, delete)
+    is logged. To enable logging for the list view, set the
+    `write_accessed_from_list_view` attribute to `True` in your `ModelAdmin`.
+    Please note that this will trigger a very intensive logging and a lots of
+    access log data will be created and stored!
+
+    Attributes:
+    write_accessed_from_list_view (bool):
+    A flag to enable/disable logging access from the list view.
+    Defaults to `False`.
+
+Example:
+
+```python
+from django.contrib import admin
+from .models import MyModel
+
+
+@admin.register(MyModel)
+class MyModelAdmin(AuditlogAdminViewAccessLogMixin, admin.ModelAdmin):
+    write_accessed_from_list_view = True  # Enable list view logging
```

Comment:
Would `enable_list_view_logging` (i.e. the comment as a variable name) be better than `write_accessed_from_list_view`? IMHO it is very clear (pro), as a con it doesn't have anything specific to do with access vs. create/update/delete. How do you see this?